src/ui/EditorNS/editor.cpp

```diff
@@ -383,9 +383,14 @@ namespace EditorNS
                 }
             }
         }
-        // If we make it here we need to make sure we're loaded.
-        waitAsyncLoad();
+        // We wait for the page to finish loading before doing this one.
         // First non-blank line
+        connect(this, &Editor::editorReady, this,
+                &Editor::detectLanguageFromContent);
```

Comment:
What is `setLanguageFromFilename` is called at some point after the `editorReady` signal was sent? Wouldn't that mean `detectLanguageFromContent` doesn't get executed?

Also, it's a bit weird for a function called `setLanguageFromFileName` to deduce the language from the doc's *contents*. IMO it should either be called differently, split into two functions, or simply removed. There are only like five languages with a non-empty `firstNonBlankLine`. Is it actually worth keeping it?